### PR TITLE
Views refactor

### PR DESCRIPTION
**Pull request by Fr3E**
Refactor of Views group

### CODE REVIEW

File: ImagePicker/ActionCell.swift [hunk 1]
```diff
@@ -1,65 +1,61 @@
...
+        updateLayout(withIndex: index, layoutConfiguration: layoutConfiguration)
         
-        let layoutModel = LayoutModel(configuration: layoutConfiguration, assets: 0)
-        let actionCount = layoutModel.numberOfItems(in: layoutConfiguration.sectionIndexForActions)
-        
-        titleLabel.textColor = UIColor.black
-        switch index {
-        case 0:
+        titleLabel.textColor = .black
+        if index == SubType.camera {
```
Comment (radimhalfar) [inline ImagePicker/ActionCell.swift:32]: may go through switch by Subtype

File: ImagePicker/AssetCell.swift [hunk 2]
```diff
@@ -162,37 +39,23 @@ class AssetCell : UICollectionViewCell, ImagePickerAssetCell {
...
+        imageView.clipsToBounds = true
+        
+        selectedImageView.frame = CGRect(x: 0, y: 0, width: 31, height: 31)
+        selectedImageView.isHidden = true
+        
+        contentView.addSubview(imageView)
+        contentView.addSubview(selectedImageView)
     }
 }
+
```
Comment (radimhalfar) [inline ImagePicker/AssetCell.swift]: keep single ending line

File: ImagePicker/AssetCell.swift [hunk 3]
```diff
@@ -162,37 +39,23 @@ class AssetCell : UICollectionViewCell, ImagePickerAssetCell {
...
-    }
-    
-    override func layoutSubviews() {
-        super.layoutSubviews()
-        foregroundView.frame = bounds
+    private func initializeViews() {
+        imageView.contentMode = .scaleAspectFill
+        imageView.clipsToBounds = true
+        
+        selectedImageView.frame = CGRect(x: 0, y: 0, width: 31, height: 31)
```
Comment (radimhalfar) [inline ImagePicker/AssetCell.swift]: CGRect(origin:, size:)

File: ImagePicker/CarvedLabel.swift [hunk 4]
```diff
@@ -63,51 +66,35 @@ final class CarvedLabel : UIView {
...
-        return CGSize(width: stringSize.width + horizontalInset*2, height: stringSize.height + verticalInset*2)
+        return CGSize(width: stringSize.width + horizontalInset * 2, height: stringSize.height + verticalInset * 2)
     }
 
     override var intrinsicContentSize: CGSize {
         return sizeThatFits(.zero)
     }
+    
+    private func drawText(_ rect: CGRect) {
+        guard let context = UIGraphicsGetCurrentContext(), text?.count ?? 0 > 0 else { return }
```
Comment (radimhalfar) [inline ImagePicker/CarvedLabel.swift]: add let textCount to guard to simplify the understanding of the condition. Make the comparison as the last one.

File: ImagePicker/RecordDurationLabel.swift [hunk 5]
```diff
@@ -78,41 +58,36 @@ final class RecordDurationLabel : UILabel {
...
     }
     
-    // MARK: Private Methods
-    
-    private var secondTimer: Timer?
-    private var indicatorTimer: Timer?
+    private func initializeViews() {
+        indicatorLayer.masksToBounds = true
+        indicatorLayer.backgroundColor = UIColor(red: 234/255, green: 53/255, blue: 52/255, alpha: 1).cgColor
+        indicatorLayer.frame.size = CGSize(width: 6, height: 6)
```
Comment (radimhalfar) [inline ImagePicker/RecordDurationLabel.swift:64]: s it possible to abstract these values into configuration ?

File: ImagePicker/VideoAssetCell.swift [hunk 6]
```diff
@@ -0,0 +1,103 @@
+// Copyright © 2018 INLOOPX. All rights reserved.
+
+import Foundation
+import Photos
```
Comment (radimhalfar) [inline ImagePicker/VideoAssetCell.swift]: Isn't foundation subset of Photos ?

File: ImagePicker/VideoAssetCell.swift [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+/// A default collection view cell that represents asset item. It supports:
+/// - shows image view of image thumbnail
+/// - icon and duration for videos
+/// - selected icon when isSelected is true
+
+final class VideoAssetCell: AssetCell {
+    static let durationFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
```
Comment (radimhalfar) [inline ImagePicker/VideoAssetCell.swift:12]: Formatter is prety much expensive to create. Extract it to static let.

**Review by radimhalfar — CHANGES_REQUESTED** [review]

File: ImagePicker/ActionCell.swift (hunk 1 above)
Comment (Fr3E) [reply ImagePicker/ActionCell.swift:32]: I think 2 case 'if' is cleaner than switch

File: ImagePicker/CarvedLabel.swift (hunk 4 above)
Comment (Fr3E) [reply ImagePicker/CarvedLabel.swift]: Done

File: ImagePicker/AssetCell.swift (hunk 3 above)
Comment (Fr3E) [reply ImagePicker/AssetCell.swift]: Done

File: ImagePicker/AssetCell.swift (hunk 2 above)
Comment (Fr3E) [reply ImagePicker/AssetCell.swift]: Done

File: ImagePicker/VideoAssetCell.swift (hunk 6 above)
Comment (Fr3E) [reply ImagePicker/VideoAssetCell.swift]: It is, changed at all places

File: ImagePicker/VideoAssetCell.swift (hunk 7 above)
Comment (Fr3E) [reply ImagePicker/VideoAssetCell.swift:12]: It is not?

File: ImagePicker/RecordDurationLabel.swift (hunk 5 above)
Comment (Fr3E) [reply ImagePicker/RecordDurationLabel.swift:64]: What configuration?

Comment (radimhalfar) [reply ImagePicker/RecordDurationLabel.swift:64]: put the configuration in enum, and use sttaic variables. the color could also be defined elsewhere in constatnts

File: ImagePicker/VideoAssetCell.swift (hunk 7 above)
Comment (radimhalfar) [reply ImagePicker/VideoAssetCell.swift:12]: 🤦‍♂️ sure it is 😂

File: ImagePicker/ActionCell.swift (hunk 1 above)
Comment (radimhalfar) [reply ImagePicker/ActionCell.swift:32]: yes, however it doesn't make sense to create enum though

**Review by radimhalfar — APPROVED** [review]

File: ImagePicker/RecordDurationLabel.swift (hunk 5 above)
Comment (Fr3E) [reply ImagePicker/RecordDurationLabel.swift:64]: Dont agree, its one shot configuration, normally would be inside xib

**Pull request merged**